web/api/v1/api.go

```diff
@@ -805,6 +808,93 @@ func rulesAlertsToAPIAlerts(rulesAlerts []*rules.Alert) []*Alert {
 	return apiAlerts
 }
 
+type metricsWithMetadata map[string]metadataSet
+
+type metadataSet map[metadata]struct{}
+
+func (ms metadataSet) MarshalJSON() ([]byte, error) {
+	var res bytes.Buffer
+	json := jsoniter.ConfigCompatibleWithStandardLibrary
+	encoder := json.NewEncoder(&res)
+
+	res.WriteByte('[')
+	for metadata := range ms {
+		if err := encoder.Encode(metadata); err != nil {
```

Comment:
How does this manage commas? (In the tests, it appears to work, but I don't understand why.)

Reply:
If my understanding of the [encoding library](https://github.com/golang/go/blob/64c9ee98b7684cf2156f620cbab4dbb6081b9771/src/encoding/json/encode.go#L729-L765) is correct, Go's JSON encoder is aware of the current state of the buffer. I know we're using `jsoniter` instead of the standard library but they claim  is 100% compatible so the heuristic should still be valid.